ls1intum/Artemis
# `Programming exercises`: Fix an issue with PlantUML diagrams if test names contain parentheses

### Checklist
#### General
#### Client

### Motivation and Context
This fixes an issue I encountered while working on #7015. Since moving and changing regular expressions at the same time makes reviewing quite hard. I decided to move this bugfix into its own small PR.

### Description
For Gradle exercises, test case names contain parentheses (e.g. `testBubbleSort()` instead of `testBubbleSort` for Maven exercises).

However, PlantUML diagrams until now did not allow the use of such parenthesis inside the test name. The whole diagram broke when trying to do so. This PR adapts the Regex to allow this and fixes the default gradle template.

### Steps for Testing

1. Create a new gradle programming exercise (to use the updated problem statement)
2. Verify that the UML diagram contains tests with parenthesis. Verify that these are correctly detected and that the correct color is shown.

### Review Progress

#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage
test case added, previously not tested.

### Screenshots
old (default template, wrong names):
![grafik](https://github.com/ls1intum/Artemis/assets/26540346/e89b7dfc-43a6-42a6-b428-ea8d2717f20d)

old with parentheses:
![grafik](https://github.com/ls1intum/Artemis/assets/26540346/84713e45-f093-4ba5-8af0-bcb530a5de01)

new:

![grafik](https://github.com/ls1intum/Artemis/assets/26540346/c6bb34b4-bb6a-4930-8e0a-ec1f5ed43499)
